src/domain/role/AppRoles.ts

```diff
@@ -0,0 +1,24 @@
+export const AppRoles: {
+    [key: string]: {
+        name: string;
+        description: string;
+        initialize: boolean;
+    };
+} = {
```

Comment:
This is moved code, but as before, having `"noUncheckedIndexedAccess": true` is masking a subpar approach. The way we are defining the type makes it type-unsafe and hinders code navigation. There are two approaches:  

- Using `satisfies Record<string, { name: string; description: string; initialize: boolean; }>` but that requires TS 4.9 (we have 4.4).  
- Defining a type alias + builder and letting TS infer the final type:  

  ```typescript
  type AppRole = { name: string; description: string; initialize: boolean };

  function role(role: AppRole): AppRole {
      return role;
  }
  ```

  And then use it:  
  ```typescript
  CONFIGURATION_ACCESS: role({ ... })
  ```